src/components/HeadCellMenu/MenuList/__tests__/MenuGroup.test.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { fireEvent, render, screen } from '@testing-library/react';
+import MenuGroup, { interceptClickOnMenuItems } from '../MenuGroup';
+import { MenuItemGroup } from '../../types';
+
+describe('MenuGroup', () => {
+  describe('interceptClickOnMenuItems', () => {
+    let menuGroups: MenuItemGroup[];
+    let cache: Record<string, any>;
+
+    test('should return proper values per menu', () => {
+      menuGroups = [
+        [{ items: [{ id: 1, icon: () => <i />, itemTitle: 'Menu#01', enabled: true }] }],
+        [{ items: [{ id: 2, icon: () => <i />, itemTitle: 'Menu#02', enabled: true }] }],
+      ];
+      expect(interceptClickOnMenuItems(menuGroups, cache)).toMatchObject(menuGroups);
+    });
+
+    test('should include onClick property if it was provided', () => {
+      menuGroups = [
+        [{ items: [{ id: 1, icon: () => <i />, enabled: true, itemTitle: 'Menu#01', onClick: () => {} }] }],
+      ];
+      const res = interceptClickOnMenuItems(menuGroups, cache);
+      expect(res[0][0].items[0].onClick).toBeTruthy;
```

Comment:
I guess you could cover the "onClick provided" case in the test above. Maybe try and click it here and test that the passed `onClick` is not the same reference and that it calls the passed in `onClick`?